### PR TITLE
Make Transcript create/update more versatile

### PR DESCRIPTION
**Pull request by Qubad786**
## [EDUCATOR-2022](https://openedx.atlassian.net/browse/EDUCATOR-2022)
Refactor `VideoTranscript.create_or_update` to be capable to update transcript file_name/file_data, language_code, file_format and provider.

### CODE REVIEW

**Comment by Qubad786** [comment]
@mushtaqak @muhammad-ammar this is ready for the review.

File: edxval/tests/test_api.py [hunk 1]
```diff
@@ -1693,10 +1693,8 @@ def setUp(self):
         self.transcript_url = api.create_or_update_video_transcript(
             self.video_id,
             'ur',
-            'The_Arrow.srt',
-            TranscriptFormat.SRT,
-            provider=TranscriptProviderType.CUSTOM,
-            file_data=File(open(self.arrow_transcript_path)),
+            metadata={'file_format': TranscriptFormat.SRT},
```
Comment (muhammad-ammar) [inline edxval/tests/test_api.py]: don't we need to pass the `provider` in the metadata?

File: edxval/tests/test_api.py [hunk 2]
```diff
@@ -1918,12 +1938,10 @@ def test_video_transcript_deletion(self):
...
         new_transcript_url = api.create_or_update_video_transcript(
-            self.video_id,
-            'ur',
-            'overwatch.srt',
-            TranscriptFormat.SRT,
-            TranscriptProviderType.CIELO24,
-            ContentFile(FILE_DATA)
+            video_id=self.video_id,
+            language_code='ur',
+            metadata=dict(provider=TranscriptProviderType.CIELO24),
```
Comment (muhammad-ammar) [inline edxval/tests/test_api.py]: don't we need to pass the`file_format` in the metadata?

File: edxval/tests/test_api.py (hunk 1 above)
Comment (Qubad786) [reply edxval/tests/test_api.py]: TranscriptProviderType.CUSTOM is default provider.

File: edxval/tests/test_api.py (hunk 2 above)
Comment (Qubad786) [reply edxval/tests/test_api.py]: Don't need to update as its already in SRT format.

File: edxval/views.py [hunk 3]
```diff
@@ -4,22 +4,24 @@
...
-from django.views.decorators.http import last_modified
 from rest_framework import generics, status
 from rest_framework.authentication import SessionAuthentication
 from rest_framework.permissions import DjangoModelPermissions
 from rest_framework.response import Response
 from rest_framework.views import APIView
 from rest_framework_oauth.authentication import OAuth2Authentication
 
-from edxval.api import (create_or_update_video_transcript,
-                        get_video_transcript, update_video_status)
```
Comment (mushtaqak) [inline edxval/views.py]: Good 👍

Comment (mushtaqak) [inline edxval/views.py]: Good 👍

**Pull request merged**